### ISSUE TITLE
Latest version of Happy DOM hangs Vitest

### ISSUE THREAD

**Issue by capricorn86** [vitest-dev/vitest#1233]
### Describe the bug

First I just want to say that this is a great library and I have recommended it to a lot of people. It is also really fun that Happy DOM is used as I am the creator of it :slightly_smiling_face: 

I just released a new major version of Happy DOM, which I believe is causing Vitest to hang in some scenarios.

In the latest major version of Happy DOM, the Window object is a [VM context](https://nodejs.org/api/vm.html#what-does-it-mean-to-contextify-an-object). This means that global keys like Object, Array, Function etc. in the Window instance are unique to the context and will not be the same as in the global object. The change was done to solve multiple problems reported to Happy DOM. 

Now when I looked at the Vitest code I noticed that the keys from the Window instance are copied to the global object. It will then override native properties (like Object, Arrray, Function etc.) with another version unique to the VM context. This will probably cause all kinds of weird behaviors.

I wish I knew that Vitest imported Happy DOM this way. I would have handled the release differently then :sweat_smile: 

Happy DOM now exports a class called GlobalWindow, which is meant to be executed within the global context.

To solve the problem either Vitest changes to use GlobalWindow, or Happy DOM changes Window to be the global window and creates a new window class called VMWindow (or similar).

Bug in Happy DOM:
https://github.com/capricorn86/happy-dom/issues/462

### Reproduction

**Reproduce global reference issues**

This will fail:
```
import { describe, expect, it } from 'vitest';

describe('suite name', () => {
    it('foo', () => {
        expect([].constructor).toBe(Array);
    });
});
```

**Reproduce Vitest hanging**
In the issue in Happy DOM several users described scenarios where just a few tests had to be run for it to happen, but I was only able to reproduce it consistently in:
https://github.com/ueberbit/lib

To reproduce:
1. git clone https://github.com/ueberbit/lib.git
2. cd ./lib
3. npm i
4. npm test

### System Info

```shell
OS: Ubuntu 21.10
Vitest version: v0.10.2
Happy DOM version: v3.1.0
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Thank you very much for your reference! We saw that we had a problem with new version of happy-dom, but didn’t have time to dig in. This will surely help us fix the issues 😄

**Comment by capricorn86** [comment]
@sheremet-va yes that should solve the issues for both versions :slightly_smiling_face: :+1:

**Comment by sheremet-va** [comment]
@capricorn86 unfortunately simple `GlobalWindow || Window` didn't help: #1234

Maybe you have other ideas why this might happen?

**Comment by davidortner-ipex** [comment]
@sheremet-va I have now debugged and apparently the import of a package called "sync-request" caused the issue. Not sure what is happening, but I will make a fix for it as soon as possible in Happy DOM.

I still suggest changing to use GlobalWindow instead of Window as other problems could emerge otherwise.

**Comment by capricorn86** [comment]
@sheremet-va I have now debugged and apparently the import of a package called "sync-request" caused the issue. Not sure what is happening, but I will make a fix for it as soon as possible in Happy DOM.

I still suggest changing to use GlobalWindow instead of Window as other problems could emerge otherwise.

**Comment by davidortner-ipex** [comment]
Weird 🤔

I only tested in the project mentioned above where it hanged and there the fix solved the problem, but I can try to debug with the test you ran as well.

**Comment by capricorn86** [comment]
Weird 🤔

I only tested in the project mentioned above where it hanged and there the fix solved the problem, but I can try to debug with the test you ran as well.

**Comment by sheremet-va** [comment]
> Weird 🤔
> 
> I only tested in the project mentioned above where it hanged and there the fix solved the problem, but I can try to debug with the test you ran as well.

Maybe it was just a fluke. We released the fix in [0.10.4](https://github.com/vitest-dev/vitest/releases/tag/v0.10.4), let's see how it will play out 😄

**Comment by capricorn86** [comment]
Yes, hopefully! :sweat_smile: It works now for the consumers that reported the problem in Happy DOM at least.
